### ISSUE TITLE
Implement better map format

### ISSUE THREAD

**Issue by samperumal** [samperumal/samperumal.github.io#10]

**Comment by antonburger** [comment]
Ideally want to keep the nice visual representation, but also allow tagging objects with values, e.g. key/door colours.

**Comment by samperumal** [comment]
Agreed. So what I was thinking was a main map file with just the symbols, and that an additional file with the metadata ( array of Json objects) either with x,y coords, or ordered by row and then column (so the first door you encounter when parsing is also the first door object in that metadata file. Thoughts?

**Comment by stankiewiczm** [comment]
@samperumal: is this now resolved, or is the current json addendum sufficient for the time being?

**Comment by mjrmua** [comment]
Suggest remove the x,y coordinates and use metadata as a key to the symbols in map.txt
#####
#  r  c#
#.r...  #
#####

{
    "#": { type = "wall", tt="#", },
    ".": { type= "floor", tt="."},
    "r": { type="rat", tt= /* etc */
}

**Comment by mjrmua** [comment]
If they have the same properties, you don't need to
if they have different properties then you give them different symbols
